### PR TITLE
Fix radial menu center text being offcenter

### LINKED ISSUE #5255: Signaling agent command GUI text is no longer centered
Just saw in a Thrive let's play that the text for "signal to emit" is not centered anymore (likely a regression with Godot 4):

kuva

I'm just quickly opening this issue. This should be extremely simple to fix by just setting the text alignment property back to centered for that one label.

Edit:
the selection pointer is also not handling correctly, but it looks like it is centered around the central text so hopefully that will automatically get fixed as well when fixing the text.
kuva

### PR DESCRIPTION
**Pull request by CI09**
**Brief Description of What This PR Does**

This PR does some stuff...

**Related Issues**
 
 - Closes #5255 

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

**Comment by CI09** [comment]
I also found a bug where indicator briefly flashes near center when cursor leaves center deadzone, if anyone knows a fix it can be pushed to this branch.

**Review by ender-titan1 — APPROVED** [review]
Tested and it appears to work.

File: src/gui_common/RadialMenu.cs [hunk 1]
```diff
@@ -96,6 +96,8 @@ public string CenterText
     public override void _Ready()
     {
         centerLabel = GetNode<Label>(CenterLabelPath);
+        centerLabel.HorizontalAlignment = HorizontalAlignment.Center;
```
Comment (hhyyrylainen) [inline src/gui_common/RadialMenu.cs:99]: Why is this required if the scene already sets this alignment?

File: src/gui_common/RadialMenu.cs [hunk 2]
```diff
@@ -332,7 +334,6 @@ private void UpdateIndicator()
         }
 
         var center = Size / 2;
-        var indicatorOffset = center - new Vector2(IndicatorSize / 2.0f, IndicatorSize);
```
Comment (hhyyrylainen) [inline src/gui_common/RadialMenu.cs:335]: What's the reasoning behind modifying this math?

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/gui_common/RadialMenu.cs (hunk 2 above)
Comment (CI09) [reply src/gui_common/RadialMenu.cs:335]: Honestly, I don't know. It just worked. Probably Godot update caused indicator center to be already centered without need to set offset or something like that.

Comment (hhyyrylainen) [reply src/gui_common/RadialMenu.cs:335]: If having this math doesn't hurt, I'd prefer to keep it to make sure it works in the future as well (and just doesn't luckily end up with the correct position).

Comment (CI09) [reply src/gui_common/RadialMenu.cs:335]: Well, it did hurt so...

Comment (hhyyrylainen) [reply src/gui_common/RadialMenu.cs:335]: Alright then, I'll check this without that code being in.

Comment (hhyyrylainen) [reply src/gui_common/RadialMenu.cs:335]: It doesn't seem like the indicator lines up correctly...

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
The duplicate centered set is now gone but the indicator isn't lined up correctly, here my cursor is pointing directly down but the indicator is not pointing exactly along the line towards the bottom of the screen:

**Comment by CI09** [comment]
Might be hacky but should do.

**Review by hhyyrylainen — APPROVED** [review]

**Comment by hhyyrylainen** [comment]
Not sure if this was a thing before but I noticed the indicator starts off at the center of the screen before jumping to the right position, so I opened a follow up issue: https://github.com/Revolutionary-Games/Thrive/issues/5271

**Pull request merged**